cachelib/cachebench/workload/KVReplayGenerator.h

```diff
@@ -97,15 +100,64 @@ class KVReplayGenerator : public ReplayGeneratorBase {
     for (uint32_t i = 0; i < numShards_; ++i) {
       stressorCtxs_.emplace_back(std::make_unique<StressorCtx>(i));
     }
+    if (!binaryFileName_.empty()) {
+      makeBinaryFile_ = true;
+    }
+    folly::Latch latch(1);
+    if (makeBinaryFile_) {
+      XLOGF(INFO,
+            "Started generating binary file from KVReplayGenerator"
+            "(amp factor {}, # of stressor threads {}, fast foward {})",
+            ampFactor_, numShards_, fastForwardCount_);
+      int fd = open(binaryFileName_.c_str(), O_RDWR | O_CREAT | O_TRUNC,
+                    S_IRUSR | S_IWUSR);
+      // first get the number of requests
+      size_t nreqs = traceStream_.getN() * ampFactor_;
+      totalReqs_ = nreqs;
+      size_t binReqSize = sizeof(BinaryRequest);
+      size_t totalSize = sizeof(size_t) + nreqs * binReqSize +
+                         nreqs * 80; // 80 since avg key size = 63
+      size_t totalSizeP = totalSize + (4096 - totalSize % 4096); // page align
+      ftruncate(fd, totalSizeP);
+      void* memory =
+          mmap(nullptr, totalSizeP, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);
+      void* copy = memory;
+      // the first entry is the total number of requests in the file, so we can
+      // calculate the key offset
+      size_t* first = reinterpret_cast<size_t*>(memory);
+      std::memcpy(first, &nreqs, sizeof(size_t));
+      first++;
+      memory = reinterpret_cast<void*>(first);
+      outputStreamReqs_ = reinterpret_cast<BinaryRequest*>(memory);
+      outputStreamKeys_ = reinterpret_cast<char*>(memory);
+      outputStreamKeys_ += nreqs * binReqSize;
+      folly::setThreadName("cb_binary_gen");
+      traceStream_.fastForwardTrace(fastForwardCount_);
+      genRequests(latch);
```

Comment:
why do you need to pass the latch here?

Reply:
In generating the binary file I don't need the latch, but in the other path (reading the CSV file normally) we use the latch to wait for preloading some requests. So it's not used in binary trace generation but I still pass it because it's a parameter to genRequests. 

Is there a better way to handle this passing the latch in this case? genRequests does require it (counts down once we have hit a number of requests).